### PR TITLE
Add `theme` prop to Source React components

### PR DESCRIPTION
**Pull request by oliverabrahams**
## What are you changing?
We are updating the implementation of themes in Source React components following a request to support [dark mode in DCAR](https://github.com/guardian/dotcom-rendering/issues/9277). This involves a few changes to the recommended way themes are handled in our Source React components: 

- **Introduction of a `theme` prop:** We are introducing a new `theme` prop in Source React components. This will allow consumers to apply a custom theme to any component. This approach is necessary because the `ThemeProvider` in Emotion is not compatible with Islands in their current form.

- **Support and deprecation plan for `ThemeProvider`:** While we will continue to support `ThemeProvider`, we plan to phase out the old themes. Instead, we will adopt new theme objects that align more closely with the Source components in Figma.

- **Reducing dependency on Emotion:** These changes will also help in gradually moving Source React components away from Emotion.

- **Discouraging use of `cssOverrides`:** We want to move away from using `cssOverrides` as the recommended method for altering these components. 

Previously, setting a theme for components like `ChoiceCard` was done using `ThemeProvider`:

```tsx
<ThemeProvider theme={myTheme}>
  <ChoiceCard>
    Select
  </ChoiceCard>
</ThemeProvider>
```

Updated way of providing a theme. Any variables you supply in the object are optional. You can override one or multiple of the defaults 

```tsx
const myTheme: ThemeChoiceCard = {
  backgroundSelected: palette.brand[500]
};
<ChoiceCard theme={myTheme}>
  Select
</ChoiceCard>
```

### CODE REVIEW

**Review by oliverabrahams — COMMENTED** [review]
Still need to update the components to use the mergeTheme function and add consistency to the colours of the stories

**Review by oliverabrahams — COMMENTED** [review]
~~Need to move all transformation functions into the ./theme.ts file~~

**Review by oliverabrahams — COMMENTED** [review]
Need to add theming to choice card group component

**Comment by oliverabrahams** [comment]
Add Theme exports to index.ts

File: libs/@guardian/source-react-components/src/choice-card/ChoiceCardGroup.stories.tsx [hunk 1]
```diff
@@ -351,3 +351,58 @@ UnControlledSingleSelectDefaultTheme.args = {
 	multi: false,
 };
 UnControlledSingleSelectDefaultTheme.storyName = 'Un-controlled Single Select';
+
+// *****************************************************************************
+
+export const WithSupportingCustomTheme: StoryFn<typeof ChoiceCardGroup> = (
```
[the remark this answers is not in the document]
Comment (jamesmockett) [reply libs/@guardian/source-react-components/src/choice-card/ChoiceCardGroup.stories.tsx:357]: The example, or the duplication of the components in the story? We could update it like the checkbox and radio group examples to support passing separate themes to the parent and child components.

**Review by jamesmockett — APPROVED** [review]
🎨 🚀

**Pull request merged**